packages/quantic/force-app/main/default/lwc/quanticTabBar/e2e/quanticTabBar.e2e.ts

```diff
@@ -0,0 +1,178 @@
+    test.describe("when the container's width can fit all the tabs", () => {
+      test('should display all the tabs without displaying the dropdown list', async ({
+        tabBar,
+      }) => {
+        const tabs = tabBar.allVisibleTabs;
+
+        expect(tabs).not.toBeNull();
+        expect(await tabs.count()).toEqual(expectedNumberOfTabs);
+
+        const isMoreDropdownVisible =
+          await tabBar.dropdownContainer.isVisible();
+        expect(isMoreDropdownVisible).toBe(false);
+      });
+    });
+
+    // Test different behaviors based on the viewport width.
+    viewportTests.forEach(
+      ({
+        testDescribe,
+        viewportWidth,
+        testLabel,
+        expectedNumberOfTabsDisplayed,
+        expectedMoreButtonLabel,
+      }) => {
+        test.describe(testDescribe, () => {
+          test.use({
+            viewport: {width: viewportWidth, height: standardViewportHeight},
+          });
+          test(testLabel, async ({tabBar}) => {
+            const displayedTabs = tabBar.allVisibleTabs;
+            expect(displayedTabs).not.toBeNull();
+            expect(await displayedTabs.count()).toEqual(
+              expectedNumberOfTabsDisplayed
+            );
+
+            expect(await tabBar.dropdownContainer.isVisible()).toBe(true);
+
+            await tabBar.clickMoreButton();
+
+            const tabsInDropdown = tabBar.allDropdownOptions;
+            expect(tabsInDropdown).not.toBeNull();
+            expect(await tabsInDropdown.count()).toEqual(
+              expectedNumberOfTabs - expectedNumberOfTabsDisplayed
+            );
+
+            expect(await tabBar.moreButton.textContent()).toEqual(
+              expectedMoreButtonLabel
+            );
+          });
+        });
+      }
+    );
+
+    test.describe('when the more button is clicked', () => {
+      // Using medium width (300px), 2 tabs should be displayed and 2 should be in the dropdown.
+      test.use({
+        viewport: {width: mediumViewportWidth, height: standardViewportHeight},
+      });
+
+      test.describe('when a tab is selected from the dropdown list', () => {
+        test('should correctly select the tab and make it active', async ({
+          tabBar,
+        }) => {
+          const expectedSelectedTabLabel = 'Tab 4';
+          const expectedNumberOfDropdownOptions = 2;
+          await tabBar.clickMoreButton();
+
+          const dropdownOptionsCount = await tabBar.allDropdownOptions.count();
+          expect(dropdownOptionsCount).not.toBeNull();
+          expect(dropdownOptionsCount).toEqual(expectedNumberOfDropdownOptions);
+
+          await tabBar.clickDropdownOption(1);
+
+          const activeTab = await tabBar.activeTab.textContent();
+
+          expect(activeTab).not.toBeNull();
+          expect(activeTab).toEqual(expectedSelectedTabLabel);
+        });
+
+        test('should trigger a new search send the correct UA analytics event', async ({
```

Comment:
Merge these two tests:

`test('should make the tab active, trigger a new search and send the correct analytics event')`

You're performing the same "action" in order to test something, so test everything that should happen from the same action.